src/main.cpp

```diff
@@ -1897,13 +1897,20 @@ void UpdateCoins(const CTransaction& tx, CCoinsViewCache& inputs, CTxUndo &txund
                 undo.nVersion = coins->nVersion;
             }
         }
+        // add outputs
+        inputs.ModifyNewCoins(tx.GetHash())->FromTx(tx, nHeight);
+    }
+    else {
+        // add outputs for coinbase tx
+        // In this case call the full ModifyCoins which will do a database
+        // lookup to be sure the coins do not already exist otherwise we do not
+        // know whether to mark them fresh or not.  We want the duplicate coinbases
+        // before BIP30 to still be properly overwritten.
+        inputs.ModifyCoins(tx.GetHash())->FromTx(tx, nHeight);
```

Comment:
This complication is unnecessary for Zcash-derived chains as explained above. Just call ``inputs.ModifyNewCoins`` unconditionally (instead of ``inputs.ModifyCoins``) outside the ``if (!tx.IsCoinBase())`` block.

Reply:
Also delete the last sentence of the commit message.